windows/src/developer/TIKE/compile/MergeKeyboardInfo.pas

```diff
@@ -289,21 +291,57 @@ function TMergeKeyboardInfo.LoadKMPFile: Boolean;
       end;
 
       FKMXTempFile := TTempFileManager.Get('.kmx');
+      FJSTempFile := TTempFileManager.Get('.js');
+
       try
-        for i := 0 to High(Zip.FileNames) do
+        if FMergingValidateIds then
         begin
-          if SameText(ExtractFileExt(Zip.FileNames[i]), '.kmx') then
+          for i := 0 to FKMPInfFile.Keyboards.Count - 1 do
           begin
-            SetLength(FKMXFiles, Length(FKMXFiles)+1);
-
-            SaveMemberToFile(Zip.FileNames[i], FKMXTempFile.Name);
+            for j := 0 to High(Zip.FileNames) do
+            begin
+              // Add the KMX to FPackageKMXFileinfos
+              if SameText(Zip.FileName[j], FKMPInfFile.Keyboards[i].ID + '.kmx') then
+              begin
+                SetLength(FPackageKMXFileInfos, Length(FPackageKMXFileInfos)+1);
+                SaveMemberToFile(Zip.FileNames[j], FKMXTempFile.Name);
+                FPackageKMXFileInfos[High(FPackageKMXFileInfos)].Filename := Zip.FileNames[j];
+                GetKeyboardInfo(FKMXTempFile.Name, False, FPackageKMXFileInfos[High(FPackageKMXFileInfos)].Info, False);
+              end;
+
+              // Add the JS to FPackageJSFileInfos
+              if SameText(Zip.FileNames[j], FKMPInfFile.Keyboards[i].ID + '.js') then
+              begin
+                SetLength(FPackageJSFileInfos, Length(FPackageJSFileInfos)+1);
+                SaveMemberToFile(Zip.FileNames[j], FJSTempFile.Name);
+                FPackageJSFileInfos[High(FPackageJSFileInfos)].Filename := Zip.FileNames[j];
+
+                // For now, apply JS keyboard to all web and mobile targets
+                // Not using GetKeyboardInfo because that only handles kmx files
+                FPackageJSFileInfos[High(FPackageJSFileInfos)].Info.Targets := 'web mobile';
```

Comment:
@mcdurdin is it safer if we assume this set
```
"platformSupport": {
  "desktopWeb": "full",
  "mobileWeb": "full",
  "android": "basic",
  "ios": "basic"
}

Reply:
fwiw, this assumption leads the following packages to mark the js keyboards as supporting `andoid` and `ios` when the kmn only lists `web`.

galaxie_greek_hebrew_mnemonic
galaxie_greek_hebrew_positional
syriac_aramaic

Reply:
Investigation showed that this change was the result of the keyboard sources being included in the project for multi-keyboard packages in the release/packages folder. Projects in the release/packages folder should contain only references to the .kps and not to the .kmn files which come from a different folder. (This is currently not well documented; @LornaSIL may wish to update the keyboard repo documentation accordingly).